File: Exercism Codes/Java/Secret Handshake/HandshakeCalculator.java
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

class HandshakeCalculator {

    List<Signal> calculateHandshake(int number) {
        String binary = Integer.toBinaryString(number);
        String[] arr = binary.split("");
        int j = 0;
        List<Signal> signals = new ArrayList<Signal>();
        for(int i=arr.length-1; i>=0; i--){
            if(arr[i].equals("1")){
                switch(j){
                    case 0:
                        signals.add(Signal.WINK);
                        break;
                    case 1:
                        signals.add(Signal.DOUBLE_BLINK);
                        break;
                    case 2:
                        signals.add(Signal.CLOSE_YOUR_EYES);
                        break;
                    case 3:
                        signals.add(Signal.JUMP);
                        break;
                    case 4:
                        Collections.reverse(signals);
                        break;
                }
            }
            j++;
        }
        return signals;
    }

}

File: Exercism Codes/Kotlin/Secret Message/HandshakeCalculator.kt
object HandshakeCalculator {
    fun calculateHandshake(number: Int): List<Signal> {
        var myList : MutableList<Signal> = mutableListOf()
        if(number > 31 || number < 1) return myList
        var signals : List<Signal> = listOf(Signal.WINK, Signal.DOUBLE_BLINK, Signal.CLOSE_YOUR_EYES, Signal.JUMP)
        val bin : String = number.toString(2).reversed()
        for(i in bin.indices) {
            if(bin[i] == '1') if(i != 4) myList.add(signals[i]) else myList = myList.reversed().toMutableList()
        }

        return myList
    }
}
